src/operations/invoke_host_function.js

```diff
@@ -35,3 +46,184 @@ export function invokeHostFunction(opts) {
 
   return new xdr.Operation(opAttributes);
 }
+
+/**
+ * Returns an operation that invokes a contract function.
```

Comment:
rather than imbuing base with this app knowhow, is it worthwhile for separation to implement these function wrappers in `js-stellar-sdk/src/contract_spec.ts` ?

Reply:
I think it's still better in base because it's just a variation of the XDR rather than something related to the RPC server or Horizon. To be honest, the only reason why `ContractSpec` is in `stellar-sdk` is because it was written in TypeScript. So yeah, I do think it belongs more here.